validphys2/src/validphys/utils.py

```diff
@@ -219,3 +220,141 @@ def scale_from_grid(grid):
+
+            # write the systype treatments
+
+            # remove stat from the uncertainty definitions
+            uncertainty_definitions.pop('stat', None)
+            
+            for i, (_, sys_dict) in enumerate(uncertainty_definitions.items()):
+                # four spaces seems to be the standard format (has to be checked for other datasets than CMS_1JET_8TEV)
+                stream.write(f"{i+1}    {sys_dict['treatment']}    {sys_dict['type']}\n")
+
+    return n_sys
+
+
+def convert_new_data_to_old(path_data_yaml, path_uncertainty_yaml, path_kinematics, path_metadata, name_dataset, path_DATA=None):
+    """
+    Convert the new data format into the old data format
+    """
+
+    # open the metadata yaml file
+    with open(path_metadata) as f:
+        metadata = yaml.safe_load(f)
+
+    # open the data yaml file
+    with open(path_data_yaml) as f:
+        data = yaml.safe_load(f)
+    
+    # open the uncertainty yaml file
+    with open(path_uncertainty_yaml) as f:
+        uncertainty = yaml.safe_load(f)
+
+    # open the kinematics yaml file
+    with open(path_kinematics) as f:
+        kinematics = yaml.safe_load(f)
+    
+    # get uncertainty definitions and values
+    uncertainty_definitions = uncertainty['definitions']
+    uncertainty_values = uncertainty['bins']
+    n_sys = uncertainty_yaml_to_systype(path_uncertainty_yaml, name_dataset, write_to_file=False)
+    stats = np.array([entr['stat'] for entr in uncertainty_values])
+
+    # get data values
+    data_values = data['data_central']
+    
+    # check whether path_DATA is provided else save it in the same directory in which the uncertainty yaml file is
+    if path_DATA is None:
+        if isinstance(path_uncertainty_yaml, str):
+            path_uncertainty_yaml = pathlib.Path(path_uncertainty_yaml)
+        path_DATA = path_uncertainty_yaml.parent / f"DATA_{name_dataset}.dat"
+    else:
+        path_DATA = pathlib.Path(path_DATA) / f"DATA_{name_dataset}.dat"
+
+    kin_names = list(kinematics['bins'][0].keys())
+    kin_values = kinematics['bins']
+    # open the DATA file for writing
+    with open(path_DATA, 'w') as stream:
+        
+        # write the header: Dataset name, number of sys errors, and number of data points, whitespace separated
+        stream.write(f"{name_dataset} {n_sys} {len(data_values)}\n")
+
+        for i, (data_value, stat) in enumerate(zip(data_values, stats)):
+            cd_line = f"{i+1}\t {metadata['implemented_observables'][0]['process_type']}\t {kin_values[i][kin_names[2]]['mid']}\t {kin_values[i][kin_names[1]]['mid']}\t {kin_values[i][kin_names[0]]['mid']}\t {data_value}\t {stat}\t"
+            for j, sys in enumerate(uncertainty_values):
+                
+                for k, (sys_name, sys_val) in enumerate(sys.items()):
+                    if sys_name == 'stat':
+                        continue
+
+                    if uncertainty_definitions[sys_name]['treatment'] == "ADD":
```

Comment:
remove `if` clause, since the systematics in the new format are all stored as additive